siliconcompiler/schema/schema_obj.py

```diff
@@ -1001,6 +1004,429 @@ def get_default(self, *keypath):
+                        elif len(items) == 1:
+                            val, = items
+                            if val != 'true' and val != 'false':
+                                step = val
+                                val = True
+                        else:
+                            val = True
+                    else:
+                        if len(items) == 3:
+                            step, index, val = items
+                        elif len(items) == 2:
+                            step, val = items
+                        else:
+                            val, = items
+                else:
+                    val = remainder
+
+                msg = f'Command line argument entered: {args} Value: {val}'
+                if step is not None:
+                    msg += f' Step: {step}'
+                if index is not None:
+                    msg += f' Index: {index}'
+                self.logger.info(msg)
+
+                # Storing in manifest
+                typestr = schema.get(*keypath, field='type')
+                if typestr.startswith('['):
+                    if self.valid(*args):
+                        self.add(*args, val, step=step, index=index)
+                    else:
+                        self.set(*args, val, step=step, index=index, clobber=True)
+                else:
+                    self.set(*args, val, step=step, index=index, clobber=True)
+
+        if post_process:
+            post_process(cmdargs)
+
+        return extra_params
+
+    ###########################################################################
+    def _get_switches(self, schema, *keypath):
+        '''Helper function for parsing switches and metavars for a keypath.'''
+        # Switch field fully describes switch format
+        switch = schema.get(*keypath, field='switch')
+
+        if switch is None:
+            switches = []
+        elif isinstance(switch, list):
+            switches = switch
+        else:
+            switches = [switch]
+        switchstrs = []
+
+        # parse out switch from metavar
+        # TODO: should we validate that metavar matches for each switch?
+        for switch in switches:
+            switchmatch = re.match(r'(-[\w_]+)\s+(.*)', switch)
+            gccmatch = re.match(r'(-[\w_]+)(.*)', switch)
+            plusmatch = re.match(r'(\+[\w_\+]+)(.*)', switch)
+
+            if switchmatch:
+                switchstr = switchmatch.group(1)
+                metavar = switchmatch.group(2)
+            elif gccmatch:
+                switchstr = gccmatch.group(1)
+                metavar = gccmatch.group(2)
+            elif plusmatch:
+                switchstr = plusmatch.group(1)
+                metavar = plusmatch.group(2)
+            switchstrs.append(switchstr)
+
+        return switchstrs, metavar
+
+    ###########################################################################
+    def read_manifest(self, filename, clear=True, clobber=True):
```

Comment:
I copied it over, but the one in SC currently replies on the messy merge manifest and I didn't want to tackle that along with this. Otherwise I agree that the Chip.read_manifest should ultimately rely on this.

Reply:
Which parts of merge manifest did you remove for this version of `_read_manifest`?

Reply:
Would be good to make an issue to fix them.

Reply:
I removed the parts that allowed for partial merges which is only needed in the running code so not needed in the schema